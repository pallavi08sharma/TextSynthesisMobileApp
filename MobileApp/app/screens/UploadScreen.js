import React from 'react';
import { View, StyleSheet, Modal } from 'react-native';
import * as Progress from 'react-native-progress';
import colors from '../config/colors';
import ActivityIndicator from '../components/ActivityIndicator';

// function UploadScreen({progress = 0, visible = false}) {
function UploadScreen({visible = false}) {
  return (
    <Modal visible={visible} transparent={true}>
        <View style={styles.modalBackground}>
            <View style={styles.container}>
                <ActivityIndicator visible={visible} width={200} height={200} speed={5}/>
            </View>
        </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
      alignItems:'center',
      justifyContent: 'space-around',
      backgroundColor: '#00000040',
      height: 250,
      width: 250,
      borderRadius: 50,
      display: 'flex',
  },
  modalBackground: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    backgroundColor: '#00000040'
  },
});

export default UploadScreen;