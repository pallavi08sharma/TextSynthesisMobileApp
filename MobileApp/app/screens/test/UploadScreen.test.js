import React from 'react';
import renderer from 'react-test-renderer';
import UploadScreen from "../UploadScreen"

describe('<UploadScreen />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<UploadScreen />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<UploadScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});