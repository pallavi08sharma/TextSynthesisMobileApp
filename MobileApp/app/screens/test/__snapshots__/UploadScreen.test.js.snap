// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadScreen /> renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000040",
          "borderRadius": 50,
          "display": "flex",
          "height": 250,
          "justifyContent": "space-around",
          "width": 250,
        }
      }
    />
  </View>
</Modal>
`;
